src/datatypes/descriptor.h

```diff
@@ -0,0 +1,238 @@
+
+#ifndef MAPPING_CORE_DESCRIPTOR_H
+#define MAPPING_CORE_DESCRIPTOR_H
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <functional>
+#include <gdal.h>
+#include <optional>
+
+#include "datatypes/raster.h"
+
+class GenericOperator;
+class QueryTools;
+
+namespace rts {
+
+    class Descriptor;
+
+    using UniqueRaster = std::unique_ptr<GenericRaster>;
+
+    /**
+     * A class representing the resolution of a raster or tile. The integer fields are signed integers because
+     * negative values can be used in the context of tiles that cover space outside of the actual raster
+     * because of the fixed tile alignment at the projections origin.
+     */
+    class Resolution {
+    public:
+        Resolution();
+        Resolution(int resX, int resY);
+
+        Resolution operator+(const Resolution &other) const;
+        Resolution operator-(const Resolution &other) const;
+
+        int resX;
+        int resY;
+    };
+
+    /**
+     * Class for representing pixel scale in two elements (x and y).
+     */
+    class Scale {
+    public:
+        Scale();
+
+        Scale(double x, double y);
+
+        double x, y;
+
+    };
+
+    /**
+     * Alias for the Scale class representing the origin of a raster.
+     */
+    using Origin = Scale;
+
+
+    /**
+    * The order of returning the tile descriptors by an operator.
+    * @Spatial Operator returns one tile for raster, then advances to the next tile and starts with the first raster again.
+    * @Temporal Operator returns all tiles from a raster, then advances to return the tiles of the next raster, etc.
+    */
+    enum class ProcessingOrder {
+        Spatial = 1,
+        Temporal = 2
+    };
+
+    /**
+    * A class for the metadata saved in a Descriptor. The Descriptor class inherits DescriptorInfo.
+    * This allows only copying the metadata of a descriptor because the getRaster closure stored in the
+    * descriptor class can not be copied.
+    */
+    class DescriptorInfo {
```

Comment:
A descriptor could also contain the offset of a tile to the origin and optional min max mean values